src/RichText.js

```diff
@@ -204,23 +204,27 @@ export function textOffsetsToSelectionState({start, end}: SelectionRange,
     let selectionState = SelectionState.createEmpty();
     for (const block of contentBlocks) {
         const blockLength = block.getLength();
-        if (start !== -1 && start < blockLength) {
-            selectionState = selectionState.merge({
-                anchorKey: block.getKey(),
-                anchorOffset: start,
-            });
-            start = -1;
-        } else {
-            start -= blockLength + 1; // +1 to account for newline between blocks
+        if (start !== -1) {
```

Comment:
is -1 a magic constant meaning "unknown" or is it literally off the front of the buffer? some commentary is vital here.